### ISSUE TITLE
Code generation is locale-sensitive and thus produces illegal Java code

### ISSUE THREAD

**Issue by Vampire** [javaparser/javaparser#3018]
There are locales, where lower-case / upper-case mappings are not what you expect.
If you for example have a turkish locale (`tr`), the lower-case version of `I` is not `i` but `ı` (no dot).
In version 3.6.8 this for example caused `public` to be `publıc` and thus non-valid Java code due to `final String codeRepresentation = this.name().toLowerCase();` in the `Modifier` enum.
This changed in 3.9.0 when `Modifier` was made a node and the code representations hard-coded.

But there are other such bugs. For anything non-human-readable like this, `toLowerCase` and `toUpperCase` should always be used with an explicit locale, just as files should always be read with an explicit encoding.

Another example is `this.codeRepresentation = name().toLowerCase();` in `Primitive` which still fails to work properly in the current 3.18.0.

This small snippet for example demonstrates the problem:
```java
Locale.setDefault(Locale.forLanguageTag("tr"));
CompilationUnit cu = new CompilationUnit();
cu.addInterface("Foo").addMethod("foo").addParameter(PrimitiveType.intType(), "foo");
new PrettyPrinter().print(cu);
```

You can even reproduce (at least some?) with your own tests if you run them with turkish locale.
If you add
```xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-surefire-plugin</artifactId>
    <configuration>
        <argLine>-Duser.language=tr</argLine>
    </configuration>
</plugin>
```
to your `javaparser-core-testing/pom.xml` build plugins, the tests start to fail due to this problem.

Basically you should at least review all `toLowerCase` and `toUpperCase` calls in your code as to whether they should be better called with an explicit locale instead of `Locale.getDefault()` which is what the parameterless version does.

**Comment by koply** [comment]
+1

**Comment by MysterAitch** [comment]
Cheers @Vampire - really appreciate the report!

Is the fix as straightforward as just setting the locale to english (or root?) on all the case conversions? e.g. one of these?
```java
            //this.codeRepresentation = name().toLowerCase(Locale.ENGLISH);
            this.codeRepresentation = name().toLowerCase(Locale.ROOT);
```
For java-specific things and "internal" keywords/tokens (e.g. `int` and `final`) this seems reasonable, but I'm not sure how this might have knock-on effects in other places (e.g. some of the utility methods).

Perhaps this could be fixed in two stages -- the first for things that are definitely good to change, and then the second stage being for the more ambiguous places?

You seem more experienced with this than me, so I would welcome a PR (and/or test cases) if you would be happy to provide :+1:

**Comment by Vampire** [comment]
I'm sorry, I have too much on my plate right now to knit another PR.
But yeah, using en or root locale should be fine for the places where it is appropriate.
That means for example when calling it on technical terms like keywords and so on.

It should basically always use the local of the text actually.
If a turkish person calls `lowerCaseMe("I live in Indiana")` it should probably result in `"i live in indiana"`, not `"ı live in ındiana"`,
just like the encoding of byte-to-character conversions have to match the encoded text and not the system of the user. :-)

So if you know the locale of the text you call it on like root or en for keywords, then use the appropriate locale.
If you don't know what you call it on because you are in a utility function, you might consider having a variant where one can provide the locale to use.

**Comment by jlerbsc** [comment]
@MysterAitch i propose you to create two utility methods (toLowerCase() and toUpperCase()) in Utils class and a variant with a specified locale parameter. By default we have to use Locale.ENGLISH (i don't know the side effect of Locale.ROOT). And we can convert all calls to toLowerCase and toUpperCase to Utils.toLowerCase() or Utils.toUpperCase(). If we encounter side effects here and there we can correct it rapidly. What do you think about this proposal?
